src/nacl/bindings/crypto_sign.py

```diff
@@ -165,6 +165,31 @@ def crypto_sign_ed25519_sk_to_curve25519(secret_key_bytes):
     return ffi.buffer(curve_secret_key, curve_secret_key_len)[:]
 
 
+def crypto_sign_ed25519_sk_to_pk(secret_key_bytes):
+    """
+    Extract the public Ed25519 key from a secret Ed25519 key (encoded
+    as bytes ``secret_key_bytes``).
+
+    Raises a ValueError if ``secret_key_bytes``is not of length
+    ``crypto_sign_SECRETKEYBYTES``
+
+    :param public_key_bytes: bytes
+    :rtype: bytes
+    """
+    if len(secret_key_bytes) != crypto_sign_SECRETKEYBYTES:
+        raise exc.ValueError("Invalid secret key")
+
+    public_key_len = crypto_sign_PUBLICKEYBYTES
+    public_key = ffi.new("unsigned char[]", public_key_len)
+
+    rc = lib.crypto_sign_ed25519_sk_to_pk(public_key, secret_key_bytes)
```

Comment:
Sorry for not asking first, but did you check whether this is really faster than 
```
     public_key = secret_key_bytes[crypto_sign_SEEDBYTES:]
```

Reply:
Yes, good call. It is about 10x faster. I have changed it.

I also added `crypto_sign_ed25519_sk_to_seed`while I was at it.